### PR TITLE
update sdk 5.1

### PR DESCRIPTION
**Pull request by bucanero**

### CODE REVIEW

File: contract/src/lib.rs [hunk 1]
```diff
@@ -326,10 +317,10 @@ impl Crossword {
         for pk in public_keys {
             let puzzle = self
                 .puzzles
-                .get(&pk)
+                .get(& PublicKey::from_str(pk.as_str()).unwrap())
```
Comment (bucanero) [inline contract/src/lib.rs]:
```suggestion
                .get(&PublicKey::from_str(pk.as_str()).unwrap())
```

File: contract/Cargo.toml [hunk 2]
```diff
@@ -8,7 +8,11 @@ edition = "2018"
 crate-type = ["cdylib"]
 
 [dependencies]
-near-sdk = "4.0.0"
+near-sdk = { version = "5.1.0", features = ["legacy"] }
+bs58 = "0.4"
```
Comment (encody) [inline contract/Cargo.toml]: `near_sdk` re-exports `bs58`

File: contract/src/lib.rs [hunk 3]
```diff
@@ -71,45 +66,41 @@ pub struct Answer {
...
     status: PuzzleStatus,
-    reward: Balance,
+    reward: NearToken,
     creator: AccountId,
     dimensions: CoordinatePair,
     answer: Vec<Answer>,
 }
 
-#[derive(BorshDeserialize, BorshSerialize, Debug)]
+#[near(serializers = [borsh])]
```
Comment (encody) [inline contract/src/lib.rs]: Isn't is the default serialization behavior of `#[near]`?

File: contract/src/lib.rs [hunk 4]
```diff
@@ -1,20 +1,18 @@
...
+    ext_contract, log, Allowance, NearToken, PanicOnDefault, Promise, PromiseResult
 };
-use near_sdk::{env, near_bindgen, AccountId, PublicKey};
+use near_sdk::{env, near, AccountId, PublicKey};
 use near_sdk::{serde_json, Gas};
 use std::convert::TryFrom;
+use std::str::FromStr;
 
 // 5 Ⓝ in yoctoNEAR
 const PRIZE_AMOUNT: u128 = 5_000_000_000_000_000_000_000_000;
```
Comment (encody) [inline contract/src/lib.rs]: This should probably be of type `NearToken`.

File: contract/Cargo.toml (hunk 2 above)
Comment (bucanero) [reply contract/Cargo.toml]: fixed

File: contract/src/lib.rs (hunk 4 above)
Comment (bucanero) [reply contract/src/lib.rs]: fixed

File: contract/src/lib.rs (hunk 3 above)
Comment (bucanero) [reply contract/src/lib.rs]: fixed

**Pull request merged**